lib/exercises.rb

```diff
@@ -1,19 +1,52 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  string_hashmap = {}
+  strings.each do |string|
+    if string_hashmap[grouped_anagrams_helper(string)]
+      string_hashmap[grouped_anagrams_helper(string)] << string
+    else
+      string_hashmap[grouped_anagrams_helper(string)] = []
+      string_hashmap[grouped_anagrams_helper(string)] << string
+    end
+  end
+  return string_hashmap.values
+end
+
+def grouped_anagrams_helper(string)
+  return string.chars.sort.join
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) in worst case because of the sorting to be done
+# Space Complexity: O(1)
 def top_k_frequent_elements(list, k)
```

Comment:
You have an O(n log n) time complexity algorithm which is pretty good and O(n) space complexity because you're making a hash and an array both proportional to the size of the original list.